### PR TITLE
fix: `ComboBox` getting stuck in pressed state

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/12682

## PR Type

What kind of change does this PR introduce?

- Bugfix
 
## What is the current behavior?

- Focus visual states were often incorrect
- On keyboard-supporting platforms the last input type always stayed stuck on keyboard
- Popup routed events got propagated to the logical parent

## What is the new behavior?

Fixed

### <samp>🤖 Generated by Copilot at 326a0ec</samp>

No summary available (Limit exceeded: required to process 89091 tokens, but only 50000 are allowed per call)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs:1]: @MartinZikmund Are we able to validate those changes in a UI or runtime test ? Or maybe using programmatic api calls?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs:1]: Done!

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/4.9

**Comment by jeromelaban** [comment]
@mergifyio backport legacy/4x

**Comment by jeromelaban** [comment]
The build is available on dev feed as 4.9.42. Give it a try and let us know so we can push it to nuget.org.

**Pull request merged**